cli/dcoscli/marathon/main.py

```diff
@@ -674,15 +674,24 @@ def _clean_up_resource_definition(properties):
     :rtype: dict
     """
     clean_properties = {}
+    ignore_properties = ["deployments", "lastTaskFailure", "tasks",
+                         "tasksRunning", "tasksHealthy", "tasksUnhealthy",
+                         "tasksStaged"]
     for k, v in properties.items():
-        if v:
-            if k in ["apps", "groups"]:
-                clean_properties[k] = [_clean_up_resource_definition(v[0])]
-            # Remove fields that marathon adds. These aren't in the json-schema
-            # because a user should not specify them. When we do a `get` to see
-            # the current schema, we may see these fields, but leaving them
-            # will cause us to unnecessarily fail the schema validation.
-            elif not k.startswith("task") and k != "lastTaskFailure":
+        # Remove fields that marathon adds. These aren't in the json-schema
+        # because a user should not specify them. When we do a `get` to see
+        # the current schema, we may see these fields, but leaving them
+        # will cause us to unnecessarily fail the schema validation.
+        if v is not None and k not in ignore_properties:
```

Comment:
Can you document what you're doing here and why?  Something about lists of dictionaries?